HPCC-14727 Allow warnings when removing superfiles.

jakesmith opened the pull request:
Pass ICodeContext to transactions, so that superfile ops. can
issue warnings.
Issue warning when superfile is removed and subfile is not marked
as belonging to the superfile.

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

jamienoss committed:
HPCC-14053 Comments not always ignored when resolving macro parameter IDs

Signed-off-by: James Noss <james.noss@lexisnexis.com>

xwang2713 committed:
HPCC-14706 Remove trademark from Clienttools Windows Start Menu

ghalliday committed:
HPCC-14705 Thor rollup group loses records following skipped transform

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

g-pan committed:
HPCC-14733 DOCS:Change copyright date to 2016 for all docs

FIX HPCC-14733 DOCS:Change copyright date to 2016 for all docs
By changing the date in the Version.xml file
which gets included in all docs.

Signed-off-by: G Panagiotatos <greg.panagiotatos@lexisnexis.com>

jakesmith committed:
HPCC-14727 Allow warnings when removing superfiles.

Pass ICodeContext to transactions, so that superfile ops. can
issue warnings.
Issue warning when superfile is removed and subfile is not marked
as belonging to the superfile.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

jakesmith commented:
@AttilaVamos - please review.

In 2 commits for clarity, 1st commit is cosmetic change only, moving methods and adding virtual qualifiers.

HPCCSmoketest commented:
Automated Smoketest Build: success
Install hpccsystems-platform-community-with-plugins_6.0.0-beta1.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

richardkchapman committed:
Merge pull request #8035 from xwang2713/HPCC_14706

HPCC-14706 Remove trademark from Clienttools Windows Start Menu

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

mckellyln committed:
HPCC-14643 Add socket keepalive to thor<->eclagent conversation

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

jakesmith committed:
HPCC-14756 SOAP calls in child query need to reinit helper

The SOAP helper object was not reinitialized per child query
execution, as a consequence if the SOAP call in the child query
had used parent row data, it would continue to use the 1st row
data it was initialized for all subsequent executions of the
SOAP call.

Fix by recreating the SOAP helper object per child query start()

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
HPCC-14742 Roxie may core if keyed join limit transform references RHS row

Ensure defaultRight is set up if there is a fail transform.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Split off candidate-5.6.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-14515 Issues sorting logical files

There was previous issues sorting within the SearchResultsWidget we issues a fix for that which in turn caused issues in our DFUQueryWidget. Modified the service call to accomodate both areas. Also, "null" is returned at times for certain files which causes sorting issues. Will assign to 0 on a per row basis but not display 0.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

richardkchapman committed:
Merge pull request #8094 from miguelvazq/HPCC-14515.1.1

HPCC-14515 Issues sorting logical files

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #8091 from richardkchapman/candidate-5.4.8

HPCC-14742 Roxie may core if keyed join limit transform references RHS row

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
Merge pull request #8096 from ghalliday/issue14705

HPCC-14705 Thor rollup group loses records following skipped transform

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.4.8' into candidate-5.6.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8098 from mckellyln/hpcc-14643

HPCC-14643 Add socket keepalive to thor<->eclagent conversation

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Commit:
HPCC-14547 Add open ability to superfile tab

Within logical file details within the superfile tab there was no open ability into superfile details or number of superfiles in the tab.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

richardkchapman committed:
Merge pull request #8102 from miguelvazq/HPCC-14547.1.1

HPCC-14547 Add open ability to superfile tab

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-14776 Improve exception handling during child init exceptions

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
Merge pull request #8097 from jakesmith/hpcc-14756

HPCC-14756 SOAP calls in child query need to reinit helper

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #8104 from jakesmith/hpcc-14776

HPCC-14776 Improve exception handling during child init exceptions

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

jakesmith committed:
HPCC-14785 Throw an error on attempt to remove super that's owned

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

jakesmith committed:
HPCC-14768 Fix spurious error adding empty superfile to super

An empty superfile can be interpreted as having some default
property attributes, e.g. global vs local.
Adding it could result in spurious errors that complain that the
empty super being added mismatches the attributes of the super
file it's being added to.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #8114 from jakesmith/hpcc-14768

HPCC-14768 Fix spurious error adding empty superfile to super

Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #8111 from jakesmith/hpcc-14785

HPCC-14785 Throw an error on attempt to remove super that's owned

Reviewed-By: Attila Vamos <attila.vamos@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

jakesmith commented:
Closing to retarget 5.6
